### PR TITLE
The Boomerang Update (feat. Castlevania: Rondo of Blood, and Gurren Lagann)

### PR DESCRIPTION
**Pull request by DeityLink**
https://www.youtube.com/watch?v=sME5_tyRbkg

image
image
battlecross
holy_firetransform

Commissioned by tinysaturn, who also made the sprites.

:cl:
* rscadd: Chaplains who adhere to the Belmont religion now spawn with a box containing clothing reminiscent of Richter Belmont, and a flask of Sacred Water in their pocket. They can also throw their bible to hit people around them with it.
* rscadd: Sacred Water is a new reagent that turns Holy Water into more of itself when in contact with it. When splashed on a floor, it creates sacred flames that only burn the unholy (that is, vampires, cultists, and any other supernatural creature)
* rscadd: Players may now forge a Battle Cross out of silver at an anvil. Battle Crosses are basically boomerangs that deal bonus damage to supernatural creatures, but they only return to the hands of those of the Belmont religion.
* experiment: Boomerangs have been rewritten from scratch and now behave more like you would expect from a boomerang in a video game. They are now a lot more reliable so their damage has been lowered from 16 to 10, but they still feel really strong so I'm looking forward to feedback by players who have been on both ends of a boomerang throw.
* imageadd: Boomerangs have also gotten a re-sprite.
* imageadd: Attacking supernatural creatures with holy weapons now displays a visual effect.
* imageadd: Kamina and Simon's glasses have gotten some visual adjustments on top of proper object icons. Furthermore, you can now unleash their power by using them in your hand and throwing them.
* bugfix: Fixed the wrong icon appearing for some bibles in the religion picker interface.

### CODE REVIEW

**Comment by DeityLink** [comment]
Some new boomerang footage:

https://github.com/user-attachments/assets/780a2da2-31f4-4fc6-a58e-fb4a9fb060b6

**Comment by Eneocho** [comment]
where is the chain whip

**Comment by DeityLink** [comment]
Already in the game, it's a premium item in the piety vend.

**Comment by Eneocho** [comment]
good good
is sacred water acquirable in any way? Aside from the belmont religion i mean

**Comment by DeityLink** [comment]
not right now, might add a way for non-belmonts to get their hands on it later. part of the design is that you need some of it to restock, and if you lose all of it, you can't get any more. Can't use premium vending items since those can be restocked from cargo. 

So for now the main method might be admin prayers, which actually fits since well, chaplains.

**Comment by Eneocho** [comment]
could make it so the belmont chaplains start with the chain already, thematically fitting and all

**Comment by Kromkar** [comment]
Make a 10u vial of sacred water a premium piety vend item imo. Shouldn't gatekeep mechanics behind a single religion. Let Belmont chaps start with like a 100u flask of it or something though

**Comment by MercuriusGH** [comment]
[HYDRO STORM](https://www.youtube.com/watch?v=FHCkF7uFrbw)

**Comment by jwhitak** [comment]
The boomerang I'm told can be made for two wood, is faster than a taser, stuns longer than a taser, and even weakens the target? If this is true, probably shouldn't be released in this state.

File: code/game/objects/items/weapons/boomerang.dm [hunk 1]
```diff
@@ -1,25 +1,242 @@
...
 	starting_materials = list(MAT_WOOD = CC_PER_SHEET_WOOD * 6)
 	w_type = RECYK_WOOD
 	flammable = TRUE
 
-/obj/item/weapon/boomerang/Destroy()
+	var/mob/originator = null
+	var/thrown = FALSE
+	var/throw_mult = 1
+	var/stun = 2
+	var/weaken = 3
```
Comment (hacker-on-steroids) [inline code/game/objects/items/weapons/boomerang.dm]: lol. lmao

**Comment by DeityLink** [comment]
Lol, Tasers stun you for 20 seconds (weaken = 10). Boomerangs currently stun you for 6 seconds (weaken = 3) which I agree is a bit long, I'm gonna lower it to 2 seconds. However remember that:

* Unlike regular projectiles, boomerangs cannot hit you if you're lying down, even if the thrower clicked you precisely
* People can just catch your boomerang with an empty hand and in throw mode

Also "weaken" is what most people understand when talking about "stun" in spess. the "stun" var makes you unable to move and act, but won't knock you down, "weaken" is what knocks you down, but you can still crawl around while weakened but no longer stunned.

As for the speed they are 50% faster. As can be seen in the video. I think it's a decent trade.

File: code/game/objects/items/weapons/boomerang.dm (hunk 1 above)
Comment (DeityLink) [reply code/game/objects/items/weapons/boomerang.dm]: meanwhile tasers (which can't be grabbed or dodged by lying down) be like 
image

**Comment by SECBATON-GRIFFON** [comment]
feature request: boomerangs launched on a mass driver should return too, if they dont already

**Comment by DeityLink** [comment]
They do already, just like they do when launched out of disposals.

Disclaimer: They don't in space/low pressure environments.

File: code/game/objects/items/weapons/boomerang.dm (hunk 1 above)
Comment (hacker-on-steroids) [reply code/game/objects/items/weapons/boomerang.dm]: would making thrown chewing gum deal 120 toxin damage(blocked if the target is wearing a fake mustache or has a prime number of credits in their virtual wallet) be reasonable too?
jesus fuck man, a taser cant be made from 6 wooden planks

**Comment by SECBATON-GRIFFON** [comment]
> Disclaimer: They don't in space/low pressure environments.

i wonder why, irl too?

**Comment by angryturnip** [comment]
Now add whipping walls to find hidden cooked chickens

**Comment by boy2mantwicethefam** [comment]
> Lol, Tasers stun you for 20 seconds (weaken = 10). Boomerangs currently stun you for 6 seconds (weaken = 3) which I agree is a bit long, I'm gonna lower it to 2 seconds. However remember that:
> 
>     * Unlike regular projectiles, boomerangs cannot hit you if you're lying down, even if the thrower clicked you precisely
> 
>     * People can just catch your boomerang with an empty hand and in throw mode
> 
> 
> Also "weaken" is what most people understand when talking about "stun" in spess. the "stun" var makes you unable to move and act, but won't knock you down, "weaken" is what knocks you down, but you can still crawl around while weakened but no longer stunned.
> 
> As for the speed they are 50% faster. As can be seen in the video. I think it's a decent trade.

Boomerang meta, greyshits will trip all the officers that carry tasers from across the screen, officers will be forced to have an empty hand and grab + throw intent at all times

**Comment by DeityLink** [comment]
I've heard your concerns and adjusted the knockdown from 5-6 sec to 2-3 sec, on top of making helmet prevent the knockdown altogether, so it won't be as much of a pain in the ass for sec offs.

https://github.com/user-attachments/assets/59e03137-695d-4b14-99e3-99857a9fa85d

Today's commits come with a surprise hi-jacking by the Dai Gurren Brigade. Kamina and Simon's glasses have gotten a slight art pass, and using them in your hand now _turns them into boomerangs that fly right back upon your nose when they come back!_ (they also deal substantial brute damage but don't knock down at all)

https://github.com/user-attachments/assets/5f1eac1f-3af9-4670-b7c3-010312292fc5

The glasses' sprites can now be seen when facing north.

image

**Comment by DeityLink** [comment]
more actually Castlevania stuff to come still, on top of further tests and adjustments.

https://github.com/user-attachments/assets/7d9ee3d4-38d3-474b-89d7-5d8b7c8f7d6f

File: code/game/objects/items/weapons/boomerang.dm [hunk 2]
```diff
@@ -1,25 +1,278 @@
...
+		last_sound_loop = world.time
+		playsound(loc,sound_loop, 35, 0)
+
+/obj/item/weapon/boomerang/proc/return_check()//lets you add conditions for the boomerang to come back
+	return TRUE
+
+/obj/item/weapon/boomerang/proc/apply_status_effects(var/mob/living/carbon/C, var/minimal_effect = 0)
+	if (ishuman(C))
+		var/mob/living/carbon/human/H = C
+		if(istype(H.head, /obj/item/clothing/head/helmet))
```
Comment (hacker-on-steroids) [inline code/game/objects/items/weapons/boomerang.dm]: why not an armor check?

**Comment by DeityLink** [comment]
I'll finish that this week-end

**Comment by west3436** [comment]
I don't love that this is locked to a single religion but it's too much high quality content to pass up on. Also the sunglass update is straight up unatomic lmao but it's not a controversial change so it's fine.

**Comment by DeityLink** [comment]
bible_throw

It's more cosmetic than anything since bible throws don't really do much damage at all.

File: code/game/objects/items/weapons/boomerang.dm (hunk 2 above)
Comment (DeityLink) [reply code/game/objects/items/weapons/boomerang.dm]: I changed it to protection from stuns being granted should you be wearing any hat that provides some armor to melee or bullet damage, so you can use metal buckets as well. I prefer that to an RNG armor check.

**Pull request merged**